nx/lib/nx/defn/grad.ex

```diff
@@ -507,6 +507,62 @@ defmodule Nx.Defn.Grad do
     to_grad(t, g, cache)
   end
 
+  defp grad(:take, [t, i, axis], _ans, g, cache) do
+    axes_range = 0..(tuple_size(t.shape) - 1)//1
+
+    indices_shape =
+      axes_range
+      |> Enum.map(fn
+        ^axis -> Tuple.to_list(i.shape)
+        _ -> 1
+      end)
+      |> List.flatten()
```

Comment:
```suggestion
      |> Enum.flat_map(fn
        ^axis -> Tuple.to_list(i.shape)
        _ -> [1]
      end)
```

A similar technique can be employed below if you want.